### DISCUSSION TITLE
使用docker-compose安装，登录后报500错误

### DISCUSSION THREAD

**Discussion by a95241815** [hhyo/Archery#1892] [General]
下载release v1.9.0，未做任何配置，直接进到docker-compose文件里使用docker-compose命令运行项目，可以正常部署容器，之后进入到archery容器进行配置，执行python3 manage.py migrate命令时，报以下错误
![image](https://user-images.githubusercontent.com/15665797/192987571-13c63486-b66c-48b5-9dd4-61642fae5469.png)
之后，执行python3 manage.py dbshell<sql/fixtures/auth_group.sql，也出现报错：
![image](https://user-images.githubusercontent.com/15665797/192990597-efcf2234-7c15-47fe-9837-3a7827e62eef.png)

**Comment by GDFboy** [comment]
修改下库的编码格式，以及配置文件里面的，改成utf8

**Reply by a95241815** [reply]
好的，感謝解答

**Comment by amzshar** [comment]
Hi, 
我也遇到一樣的問題,
請問可以具體說明要修改那些路徑的檔案嗎?

**Reply by hhyo** [reply]
确认下mysql的字符集，请问主机是使用的windows吗？

**Reply by amzshar** [reply]
yes, Window + Docker Desktop

\Archery_SQL\Archery-1.9.1\src\docker-compose\mysql\my.cnf
[client]
default-character-set=utf8mb4
是否修改這個為 utf8?

**Reply by amzshar** [reply]
mysql> show variables like "%character%";
+--------------------------+----------------------------+
| Variable_name            | Value                      |
+--------------------------+----------------------------+
| character_set_client     | latin1                     |
| character_set_connection | latin1                     |
| character_set_database   | latin1                     |
| character_set_filesystem | binary                     |
| character_set_results    | latin1                     |
| character_set_server     | latin1                     |
| character_set_system     | utf8                       |
| character_sets_dir       | /usr/share/mysql/charsets/ |
+--------------------------+----------------------------+
8 rows in set (0.01 sec)

**Reply by peng19832** [reply]
你的全是latin，肯定不行，要改成utf8mb4，或者utf8

![Uploading image.png…]()mysql> show global variables like '%char%';
+--------------------------+----------------------------+
| Variable_name            | Value                      |
+--------------------------+----------------------------+
| character_set_client     | utf8mb4                    |
| character_set_connection | utf8mb4                    |
| character_set_database   | utf8mb4                    |
| character_set_filesystem | binary                     |
| character_set_results    | utf8mb4                    |
| character_set_server     | utf8mb4                    |
| character_set_system     | utf8                       |
| character_sets_dir       | /usr/share/mysql/charsets/ |
+--------------------------+----------------------------+
8 rows in set (0.03 sec)

**Reply by a95241815** [reply]
我使用单独搭建的MySQL解决了问题

**Reply by amzshar** [reply]
After altering tables to character set utf8; 後已解決. 謝謝,

**Comment by a95241815** [comment]
嗯嗯



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年10月18日(周二) 晚上8:55
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [hhyo/Archery] 使用docker-compose安装，登录后报500错误 (Discussion #1892)




 
After altering tables to character set utf8; 後已解決. 謝謝,

**Comment by hhyo** [comment]
没有win的环境，可以帮忙看一下docker-compose的配置是否有什么问题，导致外部引用的mysql配置文件没有生效，如果能够解决，补充个faq对其他用户也是很大的帮助
